### ISSUE TITLE
Is the idea behind how copy by reference done still valid

### ISSUE THREAD

**Issue by onumossn** [offirgolan/ember-data-copyable#18]
Is https://github.com/offirgolan/ember-data-copyable/blob/master/addon/mixins/copyable.js#L192 still valid with 2.18.1? I am running into an issue where, if I copy something with a relationship and unload it, it will be in a weird state and remove one side of the relationship while maintain the other on the originals.

**Comment by offirgolan** [comment]
Yeah the logic should still be valid. Can you breakpoint [here](https://github.com/offirgolan/ember-data-copyable/blob/master/addon/mixins/copyable.js#L196) or [here](https://github.com/offirgolan/ember-data-copyable/blob/master/addon/mixins/copyable.js#L198) and see if those lines are getting hit in your app?

**Comment by onumossn** [comment]
Those lines get hit. I was imagining that the method used to unload may have changed in a way that would cause copying references in that manner to cause the models to get into a bad state.

**Comment by offirgolan** [comment]
Can you replicate this issue with an earlier version of this addon?

**Comment by mupkoo** [comment]
There is a chance that this is more related to Ember Data and unloading records. Here I will quote Chris Thoburn (runspired), who explained this to me recently.

> `unloadRecord` has a few different behaviors depending on the nature of the relationships the record has, but the easiest way of conceptualizing it is “treat as if not loaded”.
> 
> - for sync relationships we consider this to be a true-state deletion
> - for async relationships we mark the relationship as needing to be reloaded
> - for records with no relationships, we eliminate the record from the cache
>
